gen-intr-lut.sh

```diff
@@ -7,12 +7,10 @@ echo     "pub fn lookup_vector(chip: &str, intr: &str) -> Option<usize> {"
 echo     "    match chip {"
 
 for intr_path in "$@"; do
-    chip="$(basename "$(dirname "$intr_path")")"
+    chip="$(basename "${intr_path%%.*}")"
     echo "        \"$chip\" => match intr {"
 
-    sed '/=> Ok(Interrupt::.\+),$/!d
-    s/ \+\(.\+\) => Ok(Interrupt::\(.\+\)),$/            "\2" => Some(\1),/' "$intr_path"
-
+    svd interrupts --no-gaps $intr_path | awk '{print "            \""toupper(substr($2, 1, length($2)-1))"\"" " => Some(" $1"),"}'
```

Comment:
I think it would make sense to add a comment on why `toupper()` is used here:  Because `svd2rust` internally [also touppers()](https://github.com/rust-embedded/svd2rust/blob/56be78729279eeebef65110c13be8d96c0b9270f/src/generate/interrupt.rs#L44) the interrupt names (was news to me!).